[1LP][RFR] Automation Fixes for CMQE

jawatts opened the pull request:
Failures in the last RC job required some test updates. 

{{ pytest: cfme/tests/containers/test_basic_metrics.py  cfme/tests/containers/test_manageiq_ansible_custom_attributes.py cfme/tests/containers/test_ad_hoc_metrics.py --use-provider ocp-v2 }}

shalomnaim1 commented on cfme/tests/containers/test_basic_metrics.py:
```diff
@@ -18,7 +21,7 @@ def test_basic_metrics(provider):
-    assert metrics_hostname, 'Could not find route for hawkular-metrics'
+    assert metrics_hostname, 'Could not find route for metrics'
     metrics_hostname = metrics_hostname.pop()['spec']['host']
     host_url = 'https://' + metrics_hostname + '/hawkular/metrics/'
```
This test requires a massive code refactoring, IMO instead of using the ```curl``` command, we can use python's ```request``` module. In addition, to validate Prometheus using ```curl``` do like the following example:
```curl -H "Authorization: Bearer <OCP token>" -k https://<Prometheus metrics route>/graph```.

shalomnaim1 commented on cfme/tests/containers/test_basic_metrics.py:
```diff
@@ -8,8 +8,11 @@
+
 
 @pytest.mark.polarion('CMP-10205')
+@pytest.mark.uncollectif(lambda provider: provider.metrics_type == 'Prometheus')
```
We can consider parameterization on this test so it would support any provider type, but on failure, it would raise an exception on specific endpoint type.

shalomnaim1 commented on cfme/tests/containers/test_basic_metrics.py:
```diff
@@ -18,7 +21,7 @@ def test_basic_metrics(provider):
-    assert metrics_hostname, 'Could not find route for hawkular-metrics'
+    assert metrics_hostname, 'Could not find route for metrics'
     metrics_hostname = metrics_hostname.pop()['spec']['host']
     host_url = 'https://' + metrics_hostname + '/hawkular/metrics/'
```
Validating Prometheus endpoint is done by sending a request to ```https://<Promethues metrics route/graph``` and it requires add of ```Authorization``` header to the request e.g: ```curl -H "Authorization: Bearer <OCP Token>" -k https:/<Prometheus metrics route>/graph```

shalomnaim1 replied:
In addition, I think using ```curl``` command here is less appropriate, I would prefer using  python's ```requests``` module

jawatts commented:
@shalomnaim1  I agree with you that this test does need a refactor, see the TODO I added on line 11 I had added already, but I do not believe this PR is the correct place to do it. For now, I am cleaning up our test run. A future PR is needed to do a complete overhaul of this test.

psav commented on cfme/tests/containers/test_ad_hoc_metrics.py:
```diff
@@ -14,13 +14,16 @@
-    logger.info("hawkular started successfully")
+    router = [router for router in provider.mgmt.o_api.get('route')[1]['items']
+              if router["metadata"]["name"] == 'hawkular-metrics' or
+              router["metadata"]["name"] == 'prometheus'].pop()
```
**REQUIRED:** This could result in an IndexError if there are no routers - We should handle this and skip the test if so.
